game/world/worldobjects.cpp

```diff
@@ -400,11 +400,7 @@ bool WorldObjects::execTriggerEvent(const TriggerEvent& e) {
     auto& t = *i;
     if(t.name()!=e.target)
       continue; // NOTE: trigger name is not unique - more then one trigger can be activated
-
-    const bool hadDelayedEvt = t.hasDelayedEvents();
     t.processEvent(e);
-    if(!hadDelayedEvt && t.hasDelayedEvents())
```

Comment:
Same code exists in `WorldObjects::load`. Since now it's Trigger responsibility to track deferred stuff, should be moved to `AbstractTrigger::load`